Added tests, data parser and data providers, created suite xml files …

tati1228 opened the pull request:
…for smoke, negative and regression test.

To be executed using mvn test command.

tati1228 committed:
Added tests, data parser and data providers, created suite xml files for smoke, negative and regression test.
To be executed using mvn test command.

voropa commented on pom.xml:
```diff
@@ -0,0 +1,80 @@
+            <artifactId>webdrivermanager</artifactId>
+            <version>5.2.0</version>
+        </dependency>
+        <dependency>
```
Why do you need this JUnit dependency, we are using TestNG instead

tati1228 replied:
Was inserted by mistake. Will be removed.

voropa commented on src/test/java/pages/ProductsPage.java:
```diff
@@ -0,0 +1,102 @@
+    }
+
+    public void addRemoveAllProducts() {
+        for (int i = 0; i < 6; i++) {
```
What if products count changes to any other number?
It's better to iterate through all actually avaliable products at the moment e.g.
for (product of getAllProductContainers()) {
product.findElement(addRemoveButton).click();
}

tati1228 replied:
Ok, will update methods.

voropa commented on src/test/java/pages/ProductsPage.java:
```diff
@@ -0,0 +1,102 @@
+        return driver.findElement(By.xpath(String.format(itemContainer, productName)));
+    }
+
+    private List<String> getExpectedOptionNames() {
```
Page object classes should not operate with any kind of Expected data. All expected test data should be stored in test classes / test data files

tati1228 replied:
Ok, will move expected data to the test class.

voropa commented on src/test/java/tests/BaseTest.java:
```diff
@@ -0,0 +1,66 @@
+            ChromeOptions options = new ChromeOptions();
+            options.addArguments("headless");
+            driver = new ChromeDriver(options);
+         //   driver = new ChromeDriver();
```
Please remove commented lines of code

tati1228 replied:
Will do.

voropa commented on src/test/java/pages/ProductsPage.java:
```diff
@@ -0,0 +1,102 @@
+        return driver.findElement(headerContainer).isDisplayed();
+    }
+
+    public void clickAddRemoveButton(int productId) {
```
It's better to find a product by name, not by index

tati1228 replied:
Will change tests to add a product by name

tati1228 committed:
Corrected based on PR comments

tati1228 commented:
Committed following changes:
Removed JUnit dependency;
Changed AddRemoveAllProducts method;
Test data is moved to the test method;
Removed commented code;
Changed method to add a product by name.
